apps/web/src/components/templates/VariableManager.tsx

```diff
@@ -0,0 +1,209 @@
+import { IEmailBlock } from '@novu/shared';
+import { useMemo, useState } from 'react';
+import { Controller, useFieldArray, useFormContext } from 'react-hook-form';
+import { parse } from '@handlebars/parser';
+import { Code, Space, Table } from '@mantine/core';
+import styled from 'styled-components';
+import { colors, Input, Switch, Text } from '../../design-system';
+import { FieldArrayProvider } from './FieldArrayProvider';
+
+interface VariableManagerProps {
+  template: string;
+  useTitle?: boolean;
+}
+
+interface VariableComponentProps {
+  index: number;
+  template: string;
+}
+
+type MustacheVariableType = 'String' | 'Array' | 'Boolean';
+
+interface IMustacheVariable {
+  type: MustacheVariableType;
+  name: string;
+  defaultValue?: string | boolean;
+  required?: boolean;
+}
+
+export const VariableComponent = ({ index, template }: VariableComponentProps) => {
+  const { control, watch } = useFormContext();
+
+  const variableName = watch(`${template}.variables.${index}.name`);
```

Comment:
When looking at the email template, we also have custom code support. For now looks like it's not watching the variables from there and only on the visual editor